src/flamenco/runtime/program/fd_bpf_loader_v3_program.c

```diff
@@ -412,6 +412,10 @@ execute( fd_exec_instr_ctx_t * instr_ctx, fd_sbpf_validated_program_t * prog ) {
 int
 process_loader_upgradeable_instruction( fd_exec_instr_ctx_t * instr_ctx ) {
   uchar const * data = instr_ctx->instr->data;
+  if( FD_UNLIKELY( !instr_ctx || !instr_ctx->instr || !instr_ctx->instr->data ) ) {
+    FD_LOG_WARNING(( "instr_ctx %p instr_ctx->instr %p instr_ctx->instr->data %p", (void *)instr_ctx, (void *)instr_ctx->instr, (void *)instr_ctx->instr->data ));
```

Comment:
Can you make the warning message better

Reply:
Check will not work. If it's really NULL then line 414 will segfault before the check. As well as inside the FD_LOG_WARNING.